### PR TITLE
Liquibase migration scripts for the Leave and Team entities created.

### PR DESCRIPTION
**Pull request by geroge1b**
Previous issue solved. Tables Leave and Team were created in the database.

### CODE REVIEW

File: src/main/resources/config/liquibase/changelog/20180831142900_added_entity_Team.xml [hunk 1]
```diff
@@ -13,21 +13,20 @@
     <property name="floatType" value="float4" dbms="postgresql, h2"/>
     <property name="floatType" value="float" dbms="mysql, oracle, mssql"/>
 
+    <property name="uuid_type" value="uuid" dbms="postgresql"/>
+
     <!--
         Added the entity Team.
     -->
-    <changeSet id="20180831142900-1" author="jhipster">
+    <changeSet id="20180831142900-1" author="bogdan">
```
Comment (bminciu) [inline src/main/resources/config/liquibase/changelog/20180831142900_added_entity_Team.xml]: You can use "george" here, and in all other similar places, as the author name. It helps future developers know who to ask for guidance if they have questions about this entity.

File: src/main/resources/config/liquibase/changelog/20180831142901_added_entity_Leave.xml [hunk 2]
```diff
@@ -29,23 +31,19 @@
                 <constraints nullable="false" />
             </column>
 
-            <column name="jhi_type" type="varchar(255)">
+            <column name="jhi_type" type="varchar(1024)">
```
Comment (bminciu) [inline src/main/resources/config/liquibase/changelog/20180831142901_added_entity_Leave.xml]: This should rather be called `type`. Generally `jhi_` is a convention prefix marking which elements are generated by JHipster.

Comment (bminciu) [reply src/main/resources/config/liquibase/changelog/20180831142901_added_entity_Leave.xml]: As this column will only hold values of the `LeaveType` enum, I think 64 chars should be enough.

File: src/main/resources/config/liquibase/changelog/20180831142901_added_entity_Leave.xml [hunk 3]
```diff
@@ -29,23 +31,19 @@
...
                 <constraints nullable="false" />
             </column>
 
-            <column name="status" type="varchar(255)">
+            <column name="status" type="varchar(1024)">
```
Comment (bminciu) [inline src/main/resources/config/liquibase/changelog/20180831142901_added_entity_Leave.xml]: Same for this column, as it's just a database representation of the `LeaveStatus` enum, we are in full control over how long the texts will be. So I think 64 might be enough for now.

In general, a smaller DB column produces better performance.

File: src/main/resources/config/liquibase/changelog/20180831142901_added_entity_Leave.xml [hunk 4]
```diff
@@ -29,23 +31,19 @@
...
                 <constraints nullable="false" />
             </column>
 
-            <column name="employee_id" type="bigint">
+            <column name="employee_id" type="${uuid_type}">
                 <constraints nullable="true" />
```
Comment (bminciu) [inline src/main/resources/config/liquibase/changelog/20180831142901_added_entity_Leave.xml]: This is not your change, but I think there's an error in the way we defined this.

A nullable `employee_id` means that a `Leave` object can be saved into the database without an `Employee`. From a business perspective it doesn't make sense, I guess. Right?

If you agree, my suggestion would then be to change this to `nullable="false"`, and to update the `Leave.java` to add the `@NotNull` accordingly.

**Comment by geroge1b** [comment]
Thanks for the comments! I have implemented the changes. 
Now I need an advice, should I make a new pull request and close the current one or is there another way of merging the changes more easily? I've just commited and pushed the branch and am about to make a new pull request here if this is the only way. 

Commands used:
git commit -a -m "Changes from review implemented"
git push -u origin 2-migration-script-leave

**Comment by bminciu** [comment]
Nice, nice. You can just go ahead and merge this to master. ;)

**Pull request merged**